### DISCUSSION TITLE
Tips for finding the source of an error

### DISCUSSION THREAD

**Discussion by sproket** [nextgenhealthcare/connect#5721] [Q&A]
I see these in my logs sometimes.

[2023-03-15 08:23:54,681]  ERROR  (transformer:?):  -- ERROR:  JavaException: java.sql.SQLException: Subquery returns more than 1 row

I can't seem to find anywhere I would log this way but there's no stack trace to give me a clue.

Any ideas to help me trace this?

TIA

**Comment by rjkroll** [comment]
You have a SQL statement or procedure in your transformer somewhere that is returning multiple values instead of the expected 1.    Should be easy to find unless you are making an abnormal amount of SQL calls somewhere.     The only thing you can really do is try to step through the SQL manually if its a Database Reader as a Source.   If its SQL being executed when receiving messages, you could look at the channel messages, find the one causing the issue, then pass those values into your SQL to try and see whats happening.    Usually in this case, there is probably a duplicate record in a table in your SQL database that is being returned twice, or you could possibly have something mapped to a record in a table twice.

**Reply by pacmano1** [reply]
Multiple rows I think you mean.  

An example might be searching for a patient by demographic information and you expect one row but you return two.  I find this most often when doing a coalesce of a ID from a CTE where I didn't realize the CTE could return more than one row (and therefore the database doesn't know which ID I am referring to).  

```sql
with patient as (select id from patients where lastname = 'test') -- let's assume returns more than one row
select coalesce(id, 0) from patient -- sql engine will complain.
```

I usually find the offending statement by looking at the SQL engine logs and backtracking.
